src/ui/public/kuery/lib/functions/geo_bounding_box.js

```diff
@@ -0,0 +1,40 @@
+import _ from 'lodash';
+import { nodeTypes } from '../node_types';
+import * as ast from '../ast';
+
+export function buildNodeParams(fieldName, params) {
+  params = _.pick(params, 'topLeft', 'bottomRight');
+  const fieldNameArg = nodeTypes.literal.buildNode(fieldName);
+  const args = _.pairs(params).map((argument) => {
+    const [ name, value ] = argument;
+    const latLon = `${value.lat}, ${value.lon}`;
+    return nodeTypes.namedArg.buildNode(name, latLon);
+  });
+
+  return {
+    arguments: [fieldNameArg, ...args],
+  };
+}
+
+export function toElasticsearchQuery(node, indexPattern) {
+  const [ fieldNameArg, ...args ] = node.arguments;
+  const fieldName = ast.toElasticsearchQuery(fieldNameArg);
+  const field = indexPattern.fields.byName[fieldName];
+  const queryParams = args.reduce((acc, arg) => {
+    const snakeArgName = _.snakeCase(arg.name);
+    acc[snakeArgName] = ast.toElasticsearchQuery(arg);
+    return acc;
```

Comment:
Seems a little overzealous to me, but sure :)

👍  af95b5d6eb26c218083a005a10d387e36c5e6652